extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/extractors/MediaCCCLiveStreamExtractor.java

```diff
@@ -3,49 +3,56 @@
 import com.grack.nanojson.JsonArray;
 import com.grack.nanojson.JsonObject;
 import org.schabi.newpipe.extractor.MediaFormat;
-import org.schabi.newpipe.extractor.MetaInfo;
 import org.schabi.newpipe.extractor.StreamingService;
 import org.schabi.newpipe.extractor.downloader.Downloader;
 import org.schabi.newpipe.extractor.exceptions.ExtractionException;
 import org.schabi.newpipe.extractor.exceptions.ParsingException;
 import org.schabi.newpipe.extractor.linkhandler.LinkHandler;
-import org.schabi.newpipe.extractor.localization.DateWrapper;
 import org.schabi.newpipe.extractor.stream.*;
 
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import java.util.Locale;
+
+import static org.schabi.newpipe.extractor.stream.AudioStream.UNKNOWN_BITRATE;
+import static org.schabi.newpipe.extractor.utils.Utils.EMPTY_STRING;
 
 public class MediaCCCLiveStreamExtractor extends StreamExtractor {
-    private JsonArray doc = null;
     private JsonObject conference = null;
     private String group = "";
     private JsonObject room = null;
 
-    public MediaCCCLiveStreamExtractor(StreamingService service, LinkHandler linkHandler) {
+    private final List<AudioStream> audioStreams = new ArrayList<>();
+    private final List<VideoStream> videoStreams = new ArrayList<>();
+    private String firstDashUrlFound = null;
+    private String firstHlsUrlFound = null;
```

Comment:
In order to improve speed of repetitive calls of `getAudioStreams`, `getVideoOnlyStreams`, `getVideoStreams`, `getHlsUrl` and `getDashMpdUrl` methods (because otherwise documents are always parsed before returning the result) and also to reduce energy consumption.

Also, I don't understand how this can be a memory leak.

Reply:
Doesn't the `StreamInfo` class already prevent repetitive calls of those methods? But yeah, I don't see how this could cause memory leaks since the extractor object is [only used locally](https://github.com/TeamNewPipe/NewPipeExtractor/blob/f0aa46b008f2256b3e86a2182d53854fce1f8cb7/extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfo.java#L59-L65) and then it will be garbage collected (the same would happen if the stream lists were local). The garbage collector could maybe kick in earlier, or it could kick in once for each stream list if memory is really low, I don't know if this would really help.

Reply:
Hmm okay.
Anyway if it's only used internally why are the ``Extractor``-classes then ``public`` and not ``protected``?